sympy/solvers/solveset.py

```diff
@@ -2487,7 +2487,89 @@ def linear_coeffs(eq, *syms, **_kw):
     raise NonlinearError('nonlinear term encountered: %s' % t)
 
 
-def linear_eq_to_matrix(equations, *symbols):
+def _linear_system(eqs, v):
+    """return, without simplification, the list of rows containing
+    coefficients of the objects in v for each equation in eqs, and
+    a list of expressions representing the "constant" terms of each
+    equation. The constant terms might have a dependency on the
+    values of v if the equations are not full expanded or if
+    a factor in a term of an equation contains but does not equal
+    an element of v.
+
+    Raises
+    ======
+
+    NonlinearError
+        A term in an expression two or more factors from v
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x, y, z
+    >>> from sympy.solvers.solveset import _linear_system, linsolve
+    >>> from sympy import Matrix
+
+    >>> _linear_system([x + y - 3, x - y - 1], [x, y])
+    ([[1, 1], [1, -1]], [-3, -1])
+
+    If the return results are going to be used to solve for v,
+    the constants will have to be negated.
+
+    >>> A, b = Matrix(_[0]), Matrix([-i for i in _[1]])
+    >>> linsolve(A.hstack(A, b))
+    {(2, 1)}
+
+    The implicit nature of the coefficients is seen in the following
+    where `x**y` depends on `x` but, not being `x`, is assumed to
+    be independent:
+
+    >>> _linear_system([z*x + 2*x + y, x**y], [x, y])
+    ([[z + 2, 1], [0, 0]], [0, x**y])
+
+    But an error will be raised if two of the specified values appear
+    as literal factors in a single term:
+
+    >>> _linear_system([x*y - 3], [x, y])
+    Traceback (most recent call last):
+    ...
+    NonlinearError: nonlinear term: x*y
+
+    The error is not raised if a factor contains a value of v in a
+    non-literal way, however, so use this function with caution:
+
+    >>> _linear_system([x*(y + 1) + 3], [x, y])
+    ([[y + 1, 0]], [3])
+    """
+    rows = []
+    cons = []
+    x = list(v)
+    for e in eqs:
+        c, t = e.as_coeff_add(*x)
+        c = [c]
+        r = [[] for _ in range(len(x))]
+        rows.append(r)
+        for ti in t:
+            ci, f = ti.as_coeff_mul()
+            hit = [0]*len(f)
+            for i, fi in enumerate(f):
+                if fi in x:
+                    hit[i] = 1
+            if not any(hit):
+                c.append(ti)
+            else:
+                if sum(hit) != 1:
+                    raise NonlinearError('nonlinear term: %s' % ti)
+                f = list(f)
+                loc = hit.index(1)
+                fi = f.pop(loc)
+                if ci is not S.One:
+                    f.insert(0, ci)
+                r[x.index(fi)].append(Mul._from_args(f))
+        cons.append(Add(*c))
+    return [[Add(*i) for i in r] for r in rows], cons
+
+
+def linear_eq_to_matrix(equations, *symbols, **flags):
```

Comment:
New use of `**kwargs` in APIs is discouraged. Keyword-only arguments are preferred:
https://peps.python.org/pep-3102/